source/differentiation/ad/ad_drivers.cc

```diff
@@ -0,0 +1,2260 @@
+      use_stored_taped_buffer_sizes = true;
+    }
+
+
+    template <typename ADNumberType>
+    void
+    TapedDrivers<
+      ADNumberType,
+      double,
+      typename std::enable_if<ADNumberTraits<ADNumberType>::type_code ==
+                              NumberTypes::adolc_taped>::type>::
+      start_taping(const typename Types<ADNumberType>::tape_index tape_index,
+                   const bool keep_independent_values)
+    {
+      if (use_stored_taped_buffer_sizes)
+        trace_on(tape_index,
+                 keep_independent_values,
+                 obufsize,
+                 lbufsize,
+                 vbufsize,
+                 tbufsize);
+      else
+        trace_on(tape_index, keep_independent_values);
+
+      // Set some other flags to their indicated / required values
+      keep_values       = keep_independent_values;
+      is_recording_flag = true;
+    }
+
+
+    template <typename ADNumberType>
+    void
+    TapedDrivers<
+      ADNumberType,
+      double,
+      typename std::enable_if<ADNumberTraits<ADNumberType>::type_code ==
+                              NumberTypes::adolc_taped>::type>::
+      stop_taping(
+        const typename Types<ADNumberType>::tape_index active_tape_index,
+        const bool                                     write_tapes_to_file)
+    {
+      if (write_tapes_to_file)
+        trace_off(active_tape_index); // Slow
+      else
+        trace_off(); // Fast(er)
+
+      // Now that we've turned tracing off, we've definitely
+      // stopped all tape recording.
+      is_recording_flag = false;
+
+      // If the keep_values flag is set, then we expect the user to use this
+      // tape immediately after recording it. There is therefore no need to
+      // invalidate it. However, there is now also no way to double-check
+      // that the newly recorded tape is indeed the active tape.
+      if (keep_independent_values() == false)
+        active_tape = Numbers<ADNumberType>::invalid_tape_index;
+    }
+
+
+    template <typename ADNumberType>
+    std::vector<typename Types<ADNumberType>::tape_index>
+    TapedDrivers<
+      ADNumberType,
+      double,
+      typename std::enable_if<ADNumberTraits<ADNumberType>::type_code ==
+                              NumberTypes::adolc_taped>::type>::
+      get_registered_tape_indices() const
+    {
+      // We've chosen to use unsigned shorts for the tape
+      // index type (a safety precaution) so we need to
+      // perform a conversion betwwen ADOL-C's native tape
+      // index type and that chosen by us.
+      std::vector<short> registered_tape_indices_s;
+      cachedTraceTags(registered_tape_indices_s);
+
+      std::vector<typename Types<ADNumberType>::tape_index>
+        registered_tape_indices(registered_tape_indices_s.size());
+      std::copy(registered_tape_indices_s.begin(),
+                registered_tape_indices_s.end(),
+                registered_tape_indices.begin());
```

Comment:
You can do this more easily by constructing straight from an iterator range:
```
  return std::vector<...> (registered_tape_indices_s.begin(), registered_tape_indices_s.end());
```
See constructor (4) here: https://en.cppreference.com/w/cpp/container/vector/vector

Reply:
Thanks for the tip!